BoxContentSDK/BoxContentSDK/Requests/BOXFolderItemsRequest.m

```diff
@@ -43,16 +45,24 @@ - (void)cancel
     }
 }
 
+- (void)setRangeStep:(NSUInteger)rangeStep
+{
+    _rangeStep = rangeStep;
+}
+
 - (NSUInteger)rangeStep
 {
-    NSUInteger rangeStep = 1000;
-    
-    // Root folder has better performance with a smaller page size
-    if ([self.folderID isEqualToString:BOXAPIFolderIDRoot]) {
-        rangeStep = 100;
+    if (_rangeStep == 0) {
+        NSUInteger rangeStep = 1000;
+
+        // Root folder has better performance with a smaller page size
+        if ([self.folderID isEqualToString:BOXAPIFolderIDRoot]) {
+            rangeStep = 100;
+        }
+        _rangeStep = rangeStep;
```

Comment:
Shouldn't set the ivar in this case, I think we should just return the non-0 value we resolve to

Reply:
done